projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -156,7 +159,10 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
   @Input()
   public set value(value: Date) {
     this._value = value;
-    this.updateMask();
+    this.onChangeCallback(value);
+    if (value) {
```

Comment:
if input has valid `value` and at some point developer set in code behind the `value` to `null` or `undefined` the input will not update. We should not check if `value` is provided. We should check this in `updateMask` method.
There are two cases there:
- if input is focused, `_isFocused === true`, if there is no `value` just return and do nothing.
- if input is not focused, `_isFocused !== true`, if there is no `value` set `inputValue` to empty string.